x-pack/plugins/lens/public/drag_drop/providers.tsx

```diff
@@ -166,6 +215,40 @@ export function RootDragDropProvider({ children }: { children: React.ReactNode }
   );
 }
 
+export function nextValidDropTarget(
+  activeDropTarget: DropTargets | undefined,
+  draggingData: [string],
+  filterElements?: (el?: DragDropIdentifier) => boolean,
+  reverse = false
+) {
+  if (!activeDropTarget) {
+    return;
+  }
+
+  const filteredTargets = [...Object.entries(activeDropTarget.dropTargetsByOrder)].filter(
+    ([, dropTarget]) => !!dropTarget && !!(filterElements ? filterElements(dropTarget) : true)
```

Comment:
Slight tweak to code style here:

```suggestion
    ([, dropTarget]) => Boolean(dropTarget) && Boolean(filterElements?.(dropTarget))
```